File: problems_list/0413.等差数列划分/0413-等差数列划分.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numberOfArithmeticSlices = function(nums) {
    var n=nums.length;
    if(n<3) return 0;
    var dp=new Array(n).fill(0);
    var ans=0;
    for(let i=2; i<n; i++){
        if(nums[i]-nums[i-1]==nums[i-1]-nums[i-2]) dp[i]=dp[i-1]+1;
        ans+=dp[i];
    }
    return ans;
};

File: problems_list/0413.等差数列划分/0413-等差数列划分.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& nums) {
    vector<int> a(nums.size(),0);
    if(nums.size()<3) return 0;
    a[0]=a[1]=0;
    int cout=0;
    for(int i=2;i<nums.size();i++){
        if(nums[i]-nums[i-1]==nums[i-1]-nums[i-2]) a[i]=a[i-1]+1;
        cout+=a[i];
    }
    return cout;
    }
};
